### PR TITLE
Bump dependencies

### LINKED ISSUE #199: Meow is only being tested on Node 16 on CI
https://github.com/sindresorhus/meow/blob/075f96f80111548ea07afb413b197070244386fe/.github/workflows/main.yml#L12-L13

Even thought the package is marked as compatible with v12

https://github.com/sindresorhus/meow/blob/075f96f80111548ea07afb413b197070244386fe/package.json#L15-L17

### PR DESCRIPTION
**Pull request by LitoMore**
I kept the `ava` at version 3. For `ava@4` our test will fail. I haven't looked into it yet. But I can give a check with another PR.

I made denpendencies updates and GitHub Actions into one PR. Please let me know if you want to separate them into two PRs.

Resolves #199 
Closes #206

### Changes

- Bump dependencies to latest
- Add Node.js 14 and 18 to CI matrix

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -167,7 +169,8 @@ const meow = (helpText, options = {}) => {
 
 	normalizePackageData(package_);
 
-	process.title = package_.bin ? Object.keys(package_.bin)[0] : package_.name;
+	// eslint-disable-next-line node/prefer-global/process
+	global.process.title = package_.bin ? Object.keys(package_.bin)[0] : package_.name;
```
Comment (sindresorhus) [inline index.js:173]:
```suggestion
	globalThis.process.title = package_.bin ? Object.keys(package_.bin)[0] : package_.name;
```

File: index.js [hunk 2]
```diff
@@ -99,6 +100,7 @@ const validateFlags = (flags, options) => {
 	}
 };
 
+/* eslint complexity: ["error", 25] */
```
Comment (sindresorhus) [inline index.js:103]: Just turn off the complexity rule.

File: index.js (hunk 1 above)
Comment (sindresorhus) [inline index.js:173]: Why not use the imported `process`?

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -10,7 +10,9 @@ jobs:
       fail-fast: false
       matrix:
         node-version:
+          - 14
           - 16
+          - 18
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-node@v2
```
Comment (sindresorhus) [inline .github/workflows/main.yml:18]:
```suggestion
      - uses: actions/checkout@v3
      - uses: actions/setup-node@v3
```

File: index.js (hunk 1 above)
Comment (LitoMore) [reply index.js:173]: This is a mistake. Let me revert it.

Comment (LitoMore) [reply index.js]: I faced some issues with `ava@4`. This is my try-out code and I forgot to change it back.

**Pull request merged**